### PR TITLE
Get the checks for a PR green.

### PR DESCRIPTION
**Pull request by adiroiban**
This updates the code so that a PR will have all the checks greens.

I am targeting the PR checks, but I hope this is also fixing the `push` checks.

It fixed the pip deps resolution and updates some links that are permanently redirected.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -21,7 +21,10 @@ deps =
     tw203: Twisted==20.3.0
     tw212: Twisted==21.2.0
     twcurrent: Twisted
+    # See https://github.com/twisted/klein/issues/486
+    twtrunk: --use-deprecated=legacy-resolver
     twtrunk: https://github.com/twisted/twisted/tarball/trunk#egg=Twisted
+
```
Comment (adiroiban) [inline tox.ini:27]:
```suggestion
```

**Comment by adiroiban** [comment]
This is ready for review.

**Review by wsanchez — APPROVED** [review]
OMG thanks

**Comment by adiroiban** [comment]
Master still failed due to coverage publishing  

https://github.com/twisted/klein/runs/2370463492#step:11:102

@wsanchez 
with the past security issue of codecov and bash uploader
are you happy to give the on-the-fly codecov.io bash script another chance, or do you prefer to use the python uploader?

**Comment by wsanchez** [comment]
> @wsanchez
> with the past security issue of codecov and bash uploader
> are you happy to give the on-the-fly codecov.io bash script another chance, or do you prefer to use the python uploader?

I haven't thought about that much in a while, since it's been working fairly well here, but I'm open to changing it.  I'm going to go ahead and merge this and also am happy to look at changing the uploader in another PR.

Super glad you fixed this doc stuff though. It's been bugging the hell out of me, but have been juggling too many balls lately.

**Comment by wsanchez** [comment]
Oh wait right already merged, heh.

**Comment by adiroiban** [comment]
> Oh wait right already merged, heh.

welcome to the club :)

**Pull request merged**